File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.py
T = int(input())
for _ in range(T):
    N=int(input())
    d = [0] *(11)   # 11 보다 작은 수
    d[1] = 1    # 1일 때 1개 (1)
    d[2] = 2    # 2일 때 2개 (1+1, 2)
    d[3] = 4    # 3일 때 4개 (1+1+1, 1+2, 2+1, 3)
    for i in range(4,11):   # 점화식으로, 앞에 세 더한 값을 더한 값이 다음 수
        d[i] = sum(d[i-3:i])
    print(d[N])

File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")
  .map(Number);

const T = input.shift();

const max = Math.max(...input);

const dp = new Array(max + 1).fill(0);

dp[1] = 1;
dp[2] = 2;
dp[3] = 4;

for (let i = 4; i < max + 1; i++) {
  dp[i] = dp[i - 1] + dp[i - 2] + dp[i - 3];
}

for (let i = 0; i < T; i++) {
  console.log(dp[input[i]]);
  //   console.log(input[i]);
}
